File: JPLearningHub_Server/src/JPLearningHub_Server/Models/TeacherModel.cs
using JPLearningHub_Server.Entities;

namespace JPLearningHub_Server.Models
{
    public class TeacherModel
    {
        public UserModel User { get; set; } = new();

        public static TeacherModel FromTeacher(Teacher teacher, string avatarUrl)
        {
            return new TeacherModel
            {
                User = UserModel.FromUser(teacher, avatarUrl),
            };
        }
    }
}

File: JPLearningHub_Client/lib/models/teacher_model.dart
import 'package:jplearninghub/models/user_model.dart';
import 'package:json_annotation/json_annotation.dart';

part 'teacher_model.g.dart';

@JsonSerializable()
class TeacherModel {
  UserModel user;

  TeacherModel({required this.user});

  factory TeacherModel.fromJson(Map<String, dynamic> json) =>
      _$TeacherModelFromJson(json);
  Map<String, dynamic> toJson() => _$TeacherModelToJson(this);
}
